### DISCUSSION TITLE
Coming next: Really soon

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [sceneview/sceneform-android#186] [Show and tell]
Hi everyone,

A little time has passed since the last release.
But be sure that it is because a really hard/big job is made by this time and is near to be pushed.

Some clues on the picture
![image](https://user-images.githubusercontent.com/6597529/138980951-bf085058-9e97-4271-98eb-716654f3a0ee.png)

**Comment by RGregat** [comment]
RIP Java

**Comment by ThomasGorisse** [comment]
And RIP Fragment

**Comment by ThomasGorisse** [comment]
RIP Sceneform Scene (not Filement Scene)

**Comment by ThomasGorisse** [comment]
RIP InstructionController.
Say hi to InfoNodes:
```
open class SearchPlaneInfoNode(val lifecycle: ArSceneLifecycle) : Node(),
    ArSceneLifecycleObserver {

    val context get() = lifecycle.context
    var textView: TextView? = null

    init {
        isSelectable = false
        worldPosition = Vector3(0.5f, 0.5f, 0.2f)
        worldRotation = Quaternion(Vector3(0.2f, 0.0f, 0.0f))

        lifecycle.coroutineScope.launchWhenCreated {
            setRenderable(
                ViewRenderable.builder()
                    .setView(context, R.layout.view_info_search_plane)
                    .await()
            ).apply {
                isShadowCaster = false
                isShadowReceiver = false
                renderPriority = 0
                collisionShape = null
            }
        }
    }

    override fun onArFrame(session: ARCoreSession, frame: Frame, camera: Camera) {
        // Show a message based on whether tracking has failed, if planes are detected, and if the user
        // has placed any objects.
        textView?.text = trackingStateText(session, frame, camera)
    }

    open var trackingStateText: (session: ARCoreSession, frame: Frame, camera: Camera) -> String =
        { _, _, camera -> camera.trackingStateText(context) }
}
```

**Comment by ThomasGorisse** [comment]
![image](https://user-images.githubusercontent.com/6597529/139053779-7b7aa084-95d9-4be0-a8cb-6bf7b12c940e.png)

**Comment by ThomasGorisse** [comment]
A joke for now but main classes are not there anymore:

![image](https://user-images.githubusercontent.com/6597529/139054277-3a2c2976-08b4-47bd-87d0-c080114199c3.png)

**Comment by ThomasGorisse** [comment]
@RGregat @grassydragon @all
I need some help for the name. Please vote:
- sceneform-kt-android
- arsceneview-android
- sceneview-android
- scene-view-android
- other
Do you think we should keep the name sceneform for coming new repository ?

**Reply by grassydragon** [reply]
Is it going to be the same library as RealityCore or not?
I think that the `sceneform` keyword may help others to find the repository but I'm not sure if it is necessary to include it in the repository name.

**Comment by ThomasGorisse** [comment]
The Node system is still there so it's not an entity based SDK like RealityCore would have been.
The main difference is that everything is made around ArSceneView and SceneView.
Sceneform name usage is good for referencing but has some Google property restrictions.

**Reply by RGregat** [reply]
Can we actually swap the name as long as parts of the code are still based on the old repository? Besides of the legal situation there are pros and cons to keep the name Sceneform

| Sceneform - Pro | Sceneform - Cons | New Name - Pro | New Name - Cons |
| --- | --- | --- | --- |
| People know what Scenefom is | Legal restrictions |  No "dirt-bag" because of the name Sceneform and past versions | We are unknown |
| Somehow a connection to old content on medium for example | Bad experiences on older versions | No restrictions from the current legal situation | We have to make some effort in the beginning to become known |

Well, that is just a quick list of things, which came to my mind. In the end, I would say a new name would help us. But the question from my first sentence is still in the room.

**Reply by ThomasGorisse** [reply]
The good thing is that anywhere we put the new repo, thanks to the great GitHub redirection system, old links to the SceneForm Maintained repo will still work. In worst case, we can add a link on top of the repo home.
But GitHub support is really great and I think there is no issue with moving repo permanently.
Note, if you didn't noticed, I asked them 2 weeks ago to remove the Google Sceneform fork link of this repo but to keep some parts (this repo direct forks) and not others. They friendly asked me what I wanted to achieve specifically.

Concerning the organization name vs user name, depending on how the repo usage evolve, we will be able to redirect it later.
But I think that's it's too soon to move to a dedicated organization for now for 2 reason: It can make the user fill like it's another AR paying SDK and the legal declaration for sponsoring is not the same.
Let me know what you think about it.

So, after googling it and reading your POVs, here is what I go for:
- repo user/name: **ThomasGorisse/sceneview-android** 
- module name: **sceneview** (only one dependency for all)
- gradle dependency: **com.gorisse.thomas:sceneview-android:1.0.0**

The dependency will only be available on GitHub Package at first (https://docs.github.com/en/packages/working-with-a-github-packages-registry/working-with-the-gradle-registry#using-a-published-package)
And on maven central when we have reach a certain number of sponsors (still waiting for your GitHub Sponsor account for adding it to the sponsorables list)
A domain name is required for maven central so I use the one I own.

**Comment by ThomasGorisse** [comment]
You can't imagine how much just seeing this appearing on my phone makes me feel "Finally!!!!"
![image](https://user-images.githubusercontent.com/6597529/139301093-1e3cecc6-a7d2-4fdb-9e12-e18074ab2301.png)

**Comment by ThomasGorisse** [comment]
One month days and nights for just this!!! But god how cool it is so see that so much work actually works.

**Reply by RGregat** [reply]
I can feel with you. I spent days to compile a library for Android (GDAL) and was so happy to see finally the version number in the logcat :P

**Comment by ThomasGorisse** [comment]
It's here : https://github.com/ThomasGorisse/sceneview-android

Still not releasable but not so fare away.
I have some issues with the PlaneRenderer not visible.

Can you have a quick first look and solve some errors you found?

**Reply by RGregat** [reply]
I can take a look at the weekend, if it's ok for you? Maybe already tomorrow!

**Comment by ThomasGorisse** [comment]
Without depth available:
![ezgif-1-340f1a5395cf](https://user-images.githubusercontent.com/6597529/140427741-1d844a85-57e8-4172-bee9-ac833457822c.gif)
Same screencast on a Depth enable phone soon.
